src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -54,6 +59,30 @@ protected PythonNodeBase()
             ArgumentLacing = LacingStrategy.Disabled;
         }
 
+        public override void OnBeforeAdd(DynamoModel dynamoModel, WorkspaceModel workspace)
```

Comment:
workspace looks unused - just thinking ahead?

Reply:
In my mind, this "life-cycle method" made more sense if you only got the workspace where the node was being added. However, you have the same problem, `WorkspaceModel` has no visibility of the `DynamoModel` associated with it. In the end I kept both because only providing the `DynamoModel` made less sense as an API, but it's true it's not really used.